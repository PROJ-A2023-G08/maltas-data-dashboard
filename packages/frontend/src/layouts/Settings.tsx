import React, { useState } from "react";
import { Formik, Form, Field, ErrorMessage, FieldProps } from "formik";
import * as Yup from "yup";
import {
  Button,
  Card,
  CardContent,
  Switch,
  TextField,
  Typography,
  Avatar,
  MenuItem,
} from "@mui/material";
import { useDropzone, Accept } from "react-dropzone";
import { useTranslation } from "next-i18next";

const acceptedFileTypes: unknown = 'image/*';

const validationSchema = Yup.object().shape({
  displayName: Yup.string().required("Display name is required"),
  email: Yup.string()
    .email("Invalid email address")
    .required("Email is required"),
  currentPassword: Yup.string().required("Current password is required"),
  newPassword: Yup.string()
    .min(6, "New password must be at least 6 characters")
    .required("New password is required"),
  confirmPassword: Yup.string()
    .oneOf([Yup.ref("newPassword"), undefined], "Passwords must match")
    .required("Confirm password is required"),
});

const ImageUpload: React.FC<{
  image: string;
  onImageChange: (image: string) => void;
}> = ({ image, onImageChange }) => {
  const [file, setFile] = useState<File | null>(null);

  const onDrop = (acceptedFiles: File[]) => {
    //todo, we need to allow only image upload
    if (acceptedFiles.length > 0) {
      setFile(acceptedFiles[0]);
      const reader = new FileReader();
      reader.onload = (e) => {
        onImageChange(e.target?.result as string);
      };
      reader.readAsDataURL(acceptedFiles[0]);
    }
  };

  const { getRootProps, getInputProps } = useDropzone({
    onDrop,
    accept: acceptedFileTypes as Accept,
  });

  return (
    <div className="flex items-center ml-20 flex-wrap">
      <Avatar
        src={file ? URL.createObjectURL(file) : image} // Display the selected image or the existing one.
        alt="Profile Photo"
        sx={{ width: 100, height: 100 }}
      />
      
      <div className="ml-4" {...getRootProps()} style={{ cursor: "pointer" }}>
        <input {...getInputProps()} />
        <Typography variant="h5" color="primary" component="div">
          Upload
        </Typography>
      </div>

      {file && <Typography className="ml-2" variant="body2">Uploading: {file.name}</Typography>}
    </div>
  );
};

const SettingsPage: React.FC = () => {
  const { t } = useTranslation("common");
  const initialValues = {
    displayName: "",
    email: "",
    currentPassword: "",
    newPassword: "",
    confirmPassword: "",
    language: "en",
    notifications: false,
    profilePhoto: "",
  };

  const handleSubmit = (values: any) => {
    // Handle form submission, e.g., update user settings
    console.log(values);
  };

  return (
    <Formik
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={handleSubmit}
    >
      {({ values, isSubmitting }) => (
        <Form>
          <Card>
            <CardContent>
              <div className="flex ">
                <Typography variant="h5">{t("settings.ProfilePhoto")}</Typography>

                <Field name="profilePhoto" label="Profile Photo">
                  {({ field }: FieldProps<string>) => (
                    <ImageUpload
                      image={field.value}
                      onImageChange={field.onChange}
                    />
                  )}
                </Field>
              </div>
              <hr className="text-gray-300 m-6 ml-12 mr-12" />
              <div>
                <Typography className="mb-4 mt-4" variant="h5">
                {t("settings.PersonalInfo")}
                </Typography>
                <div className="ml-12 md:ml-28 lg:ml-32 max-w-xl">
                  <Field
                    className="mb-4"
                    as={TextField}
                    name="displayName"
                    label={t("settings.NameField")}
                    fullWidth
                  />
                  <Field
                    className="mb-4"
                    as={TextField}
                    name="email"
                    label={t("settings.EmailField")}
                    fullWidth
                  />
                </div>
              </div>
              <hr className="text-gray-300 m-6 ml-12 mr-12" />

              <Typography className="mb-4" variant="h5">
              {t("settings.ChangePassword")}
              </Typography>
              <div className="ml-12 md:ml-28 lg:ml-32 max-w-xl">
                <Field
                  className="mb-4"
                  as={TextField}
                  type="password"
                  name="currentPassword"
                  label={t("settings.CurrentPassword")}
                  fullWidth
                />
                <Field
                  className="mb-4"
                  as={TextField}
                  type="password"
                  name="newPassword"
                  label={t("settings.NewPassword")}
                  fullWidth
                />
                <Field
                  className="mb-4"
                  as={TextField}
                  type="password"
                  name="confirmPassword"
                  label={t("settings.ConfirmPassword")}
                  fullWidth
                />
                <Button
                  type="submit"
                  className="mb-4"
                  variant="contained"
                  color="primary"
                  disabled={isSubmitting}
                >
                  {t("settings.UpdatePassword")}
                </Button>
              </div>
              <hr className="text-gray-300 m-6 ml-12 mr-12" />

              <Typography className="mb-4" variant="h5">
              {t("settings.Language")}
              </Typography>
              <div className="ml-12 md:ml-28 lg:ml-32 max-w-xl">
                <Field
                  className="mb-4"
                  as={TextField}
                  select
                  name="language"
                  label={t("settings.Language")}
                  fullWidth
                >
                  <MenuItem value="en">{t("settings.English")}</MenuItem>
                  <MenuItem value="fi">{t("settings.Finnish")}</MenuItem>
                  <MenuItem value="sv">{t("settings.Swedish")}</MenuItem>
                </Field>
              </div>
              <hr className="text-gray-300 m-6 ml-12 mr-12" />
              <Typography className="mb-4" variant="h5">
              {t("settings.Notifications")}
              </Typography>
              <div className="ml-12 md:ml-28 lg:ml-32 max-w-xl">
                <Field
                  className="mb-4"
                  as={Switch}
                  type="checkbox"
                  name="notifications"
                  checked={values.notifications}
                />
              </div>
              <hr className="bg-gray-200 divide-blue-200 m-6 ml-12 mr-12" />
              <div className="flex mt-3 justify-end">
                <Button
                  className="mb-4"
                  type="submit"
                  variant="contained"
                  color="primary"
                  disabled={isSubmitting}
                >
                  {t("settings.Save")}
                </Button>
                <Button className="mb-4 ml-4" type="button" variant="outlined">
                {t("settings.Cancel")}
                </Button>
              </div>
            </CardContent>
          </Card>
        </Form>
      )}
    </Formik>
  );
};

export default SettingsPage;
