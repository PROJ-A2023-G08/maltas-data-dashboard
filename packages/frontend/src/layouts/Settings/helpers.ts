import * as Yup from "yup";


  export const validationSchemaPersonalInfo = Yup.object().shape({
    firstname: Yup.string().required("first name is required"),
    lastname: Yup.string().required("first name is required"),
    email: Yup.string()
      .email("Invalid email address")
      .required("Email is required"),
  });
  

  export const validationSchemaUpdatePassword = Yup.object().shape({
    currentPassword: Yup.string().required("Current password is required"),
    newPassword: Yup.string()
      .min(6, "New password must be at least 6 characters")
      .required("New password is required"),
    confirmPassword: Yup.string()
      .oneOf([Yup.ref("newPassword"), undefined], "Passwords must match")
      .required("Confirm password is required"),
  });


  export const validationSchemaUserSettings = Yup.object().shape({
    language: Yup.string(),
    notifications: Yup.boolean()
  });


  export const validationSchemaUploadImage = Yup.object().shape({
    profilePhoto: Yup.string().required("Please Upload an image"),
  });


  export interface ProfilePhoto {
    profilePhoto: string;
  };

  export interface PersonalInfo {
    firstname: string;
    lastname: string;
    email: string;
  };

  export interface UpdatePassword {
    currentPassword: string;
    newPassword: string;
    confirmPassword: string;
  };

  export interface UpdateUserSettings {
    language: string;
    notifications: boolean;
  };




