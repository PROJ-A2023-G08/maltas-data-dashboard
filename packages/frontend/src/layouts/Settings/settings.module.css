.input {
    box-shadow: rgba(0, 0, 0, 0.2) 0px 0px 0px 1px;
}

.input:focus {
    outline: none;
    box-shadow: rgba(102, 178, 255, 0.5) 0px 0px 0px 1px;
  }
  