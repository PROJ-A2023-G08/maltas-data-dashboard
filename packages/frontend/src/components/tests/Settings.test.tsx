import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Settings from "../../layouts/Settings/Settings";
import "@testing-library/jest-dom";

// Mock the useTranslation hook from i18next to return a translation function
jest.mock("next-i18next", () => ({
  useTranslation: jest.fn().mockReturnValue({
    // @ts-ignore
    t: (str) => str,
  }),
}));

//U01.2
it("Settings renders correctly", () => {
  const setActiveComponent = jest.fn();
  const tree = renderer.create(<Settings />).toJSON();
  expect(tree).toMatchSnapshot();
});
