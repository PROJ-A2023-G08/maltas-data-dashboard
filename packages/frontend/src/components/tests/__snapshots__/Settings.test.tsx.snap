// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders correctly 1`] = `
[
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div>
          <h5
            className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            Profile Photo
          </h5>
          <div
            className="ml-12 md:ml-28 lg:ml-32 max-w-xl"
          >
            <div
              className="flex items-center flex-wrap"
            >
              <div
                className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-hidgkg-MuiAvatar-root"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
              <div
                className="ml-4"
                onBlur={[Function]}
                onClick={[Function]}
                onDragEnter={[Function]}
                onDragLeave={[Function]}
                onDragOver={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="presentation"
                style={
                  {
                    "cursor": "pointer",
                  }
                }
                tabIndex={0}
              >
                <input
                  multiple={true}
                  name="profilePhoto"
                  onChange={[Function]}
                  onClick={[Function]}
                  style={
                    {
                      "display": "none",
                    }
                  }
                  tabIndex={-1}
                  type="file"
                />
                <div
                  className="MuiTypography-root MuiTypography-h5 css-1invrju-MuiTypography-root"
                >
                  Upload
                </div>
              </div>
            </div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium mb-4 mt-3 css-sghohy-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              Update Profile Image
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>,
  <hr
    className="text-gray-300 m-6 ml-12 mr-12"
  />,
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div>
          <h5
            className="MuiTypography-root MuiTypography-h5 mb-4 mt-4 css-ag7rrr-MuiTypography-root"
          >
            settings.PersonalInfo
          </h5>
          <div
            className="ml-12 md:ml-28 lg:ml-32 max-w-xl"
          >
            <div
              className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root mb-2 css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={false}
                htmlFor=":r0:"
                id=":r0:-label"
              >
                First Name
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  id=":r0:"
                  name="firstname"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-yjsfm1"
                  >
                    <span>
                      First Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root mb-4 css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={false}
                htmlFor=":r1:"
                id=":r1:-label"
              >
                Last Name
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  id=":r1:"
                  name="lastname"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-yjsfm1"
                  >
                    <span>
                      Last Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root mb-4 css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={true}
                htmlFor=":r2:"
                id=":r2:-label"
              >
                settings.EmailField
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={true}
                  id=":r2:"
                  name="email"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value="myemail@gmail.com"
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-14lo706"
                  >
                    <span>
                      settings.EmailField
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium mb-4 css-sghohy-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              Update Personal Information
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>,
  <hr
    className="text-gray-300 m-6 ml-12 mr-12"
  />,
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 mb-4 css-ag7rrr-MuiTypography-root"
        >
          Change Password
        </h5>
        <div
          className="ml-12 md:ml-28 lg:ml-32 max-w-xl"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root mb-2 css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r3:"
              id=":r3:-label"
            >
              Current Password
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r3:"
                name="currentPassword"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Current Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root mb-2 css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r4:"
              id=":r4:-label"
            >
              New Password
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r4:"
                name="newPassword"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    New Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root mb-2 css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r5:"
              id=":r5:-label"
            >
              Confirm Password
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r5:"
                name="confirmPassword"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Confirm Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium mb-4 css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            Update Password
          </button>
        </div>
      </div>
    </div>
  </form>,
  <hr
    className="text-gray-300 m-6 ml-12 mr-12"
  />,
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 mb-4 css-ag7rrr-MuiTypography-root"
        >
          Language
        </h5>
        <div
          className="ml-12 md:ml-28 lg:ml-32 max-w-xl"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root mb-4 css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={true}
              htmlFor=":r6:"
              id=":r6:-label"
            >
              Language
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <div
                aria-controls=":r7:"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":r6:-label :r6:"
                className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r6:"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="combobox"
                tabIndex={0}
              >
                English
              </div>
              <input
                aria-hidden={true}
                aria-invalid={false}
                autoFocus={false}
                className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                disabled={false}
                name="language"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value="en"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-14lo706"
                >
                  <span>
                    Language
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <h5
          className="MuiTypography-root MuiTypography-h5 mb-4 css-ag7rrr-MuiTypography-root"
        >
          Notifications
        </h5>
        <div
          className="ml-12 md:ml-28 lg:ml-32 max-w-xl"
        >
          <div
            className="w-full"
          >
            <span
              className="MuiSwitch-root MuiSwitch-sizeMedium mb-4 css-julti5-MuiSwitch-root"
            >
              <span
                className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                  name="notifications"
                  onChange={[Function]}
                  required={false}
                  type="checkbox"
                  value={false}
                />
                <span
                  className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                />
              </span>
              <span
                className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
              />
            </span>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium mb-4 css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            Update User Settings
          </button>
        </div>
      </div>
    </div>
  </form>,
]
`;
